mp/src/game/server/momentum/tickset.cpp

```diff
@@ -96,7 +123,51 @@ bool TickSet::SetTickrate(Tickrate trNew)
     return false;
 }
 
-static void OnTickRateChange(IConVar *var, const char* pOldValue, float fOldValue)
+
+static void OnTickRateSet(const CCommand &command)
+{
+    constexpr unsigned rateCount = sizeof(TickSet::s_DefinedRates) /
+        sizeof(*TickSet::s_DefinedRates);
+
+    // Search defined rates for one with a string matching the command argument.
+    for (unsigned rateI = 0; rateI < rateCount; rateI++)
+    {
+        if (!strcmp(command.ArgS(), TickSet::s_DefinedRates[rateI].sType))
+        {
+            TickSet::SetTickrate(TickSet::s_DefinedRates[rateI].fTickRate);
+            return;
+        }
+    }
+    Warning("Unknown tickrate. Use sv_interval_per_tick to set custom tickrates.");
+}
+
+static int OnTickRateAutocomplete(const char* partial,
+    char commands[COMMAND_COMPLETION_MAXITEMS][COMMAND_COMPLETION_ITEM_LENGTH])
+{
+    constexpr unsigned rateCount = sizeof(TickSet::s_DefinedRates) /
+        sizeof(*TickSet::s_DefinedRates);
+
+    // Move start of the string to expected start of parameter. (null char counted for space)
+    partial += sizeof("sv_tickrate");
+
+    const unsigned partialLength = strlen(partial);
+    unsigned suggestionCount = 0;
+
+    // Search defined rates for one with a string matching the command argument.
+    for (unsigned rateI = 0; rateI < rateCount && rateI < COMMAND_COMPLETION_MAXITEMS; rateI++)
+    {
+        if (!strncmp(partial, TickSet::s_DefinedRates[rateI].sType, partialLength))
```

Comment:
Consider listing them all if the user is typing just `sv_tickrate ` or `sv_tickrate`:
```suggestion
        if ((!partial || !partial[0]) || !Q_strncmp(partial, TickSet::s_DefinedRates[rateI].sType, partialLength))
```

Reply:
The callback isn't called until the space is entered, and the current code displays all rates once the space is entered.

Reply:
Annoyingly, the callback also isn't called if you autocomplete sv_tickrate, which leaves you with a space and no suggestions.

Reply:
If the user is entering `sv_tickrate `, `partial` would be an empty string. An empty string evaluates to the `\0` character, so doing `Q_strncmp` against an actual tickrate: "88" vs "\0" would cause this code to not actually add any suggestions.

In my suggestion I do `!partial[0]` which would check that they haven't entered anything, which then goes and adds everything as a suggestion. Them typing, or using arrow keys to get a suggestion, etc, would then make partial's first character be non-null, and the second part (the `Q_strncmp`) would take over.